.github/workflows/generic_test.yml

```diff
@@ -14,6 +14,9 @@ jobs:
   run-tests:
     name: 'Test'
     runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        part: [0, 1, 2, 3]
```

Comment:
Is this only to support the `make generate-accounts` for tests below?

Is that an issue for running tests locally in this manner too? I'd need to review the Github Actions docs. It's not clear to me if you're configuring the CI here to run the matrix on the same runner, or separate runners per `part`. I assume it's the latter.

Thus we're splitting the testing out to 4 runners / machines instead of the present 1 runner on a single machine? And then 1 of those will run all serial tests, while the other 3 run two parallel tests at a time?

Our CI has introduced some concurrency here in addition to parallelism? (_I think the technically correct term is the CI is providing parallelism, and BATS jobs is providing concurrency, but whatever_ :man_shrugging: :stuck_out_tongue: )

Reply:
> Thus we're splitting the testing out to 4 runners / machines instead of the present 1 runner on a single machine? And then 1 of those will run all serial tests, while the other 3 run two parallel tests at a time?

Yes

> Our CI has introduced some concurrency here in addition to parallelism?

Indeed :D I think this is not a problem though, as there are no write-after-x conflicts, because we do not write to a cache. Or de we? In the `generic-test`, there is no `cache-to: ..`, so all is well?

---

> Think the technically correct term is the CI is providing parallelism, and BATS jobs is providing concurrency

This is absolutely correct.